confluent-kafka-plugins/src/main/java/io/cdap/plugin/confluent/streaming/source/ConfluentStreamingSourceConfig.java

```diff
@@ -188,12 +193,12 @@ public class ConfluentStreamingSourceConfig extends ReferencePluginConfig implem
   @Nullable
   private final String schemaRegistryApiSecret;
 
-  @Name(NAME_VALUE_FIELD)
-  @Description("Name of the field containing the message payload. Required when Schema Registry is used." +
-    "This field will be used to infer schema from Schema Registry.")
+  @Name(NAME_VALUE_SCHEMA)
```

Comment:
Is this a backwards incompatible change? If we have a pipeline with older version of this plugin, will the existing schema field get ignored?

Reply:
This property has just been renamed to Value Schema to guide the user to select the value schema name and not any random field name present in the schema.

Before the fix, the user would enter the field name and not the schema name. All the fields in the schema would still get added to the final output schema but the schema name was displayed as the field name entered by the user.

After the fix, the final output schema will be formed with the correct schema name containing all the fields.

Reply:
This could be an issue for upgrade.

Reply:
Yes, some issues have been observed during the upgrade.

On upgrading the application, the following response was seen
```
java.lang.IllegalArgumentException: Failed to configure pipeline: Stage 'Confluent Kafka' encountered : Message Value Schema should be provided when Schema Registry is used
Caused by: io.cdap.cdap.etl.api.validation.ValidationException: Errors were encountered during validation. Message Value Schema should be provided when Schema Registry is used.
```

And when `valueSchema` variable was renamed back to `valueField` (label still showing Value Schema), the application upgrade resulted in the following error
```
java.lang.IllegalArgumentException: Failed to configure pipeline: Stage 'Confluent Kafka' encountered : Invalid value schema name 'my_field1'.
Caused by: io.cdap.cdap.etl.api.validation.ValidationException: Errors were encountered during validation. Invalid value schema name 'my_field1'.
```
This error came because `my_field1` is not a valid schema name in the Schema Registry. 
**Note:** `my_field1` was the value specified in `valueField` property

The application upgrade was successful when I removed the following `if` condition, and then 
```
if (valueSchema.getRecordName().equals(conf.getvalueField())) {
          newFields.add(Schema.Field.of(conf.getvalueField(), valueSchema));
        } else {
          failureCollector.addFailure(String.format("Invalid value schema name '%s'.", conf.getvalueField()),
                                      null)
            .withConfigProperty(ConfluentStreamingSourceConfig.NAME_VALUE_FIELD);
          throw failureCollector.getOrThrowException();
        }
```
and just added this
`newFields.add(Schema.Field.of(conf.getvalueField(), valueSchema));`

But this means that the user can enter any text and there is no validation to check whether it is a valid schema name or not. I haven't committed the code changes for this yet. We can go over this in today's sync-up.

Reply:
I've renamed the variable back to `valueField` and also removed the schema name validation check. 0ddbb7b903ee00ac5eddb5a12793df5585c04b22
Tested upgrading the application with the latest plugin version and did not observe any issues.